server/src/domain/person/person.service.ts

```diff
@@ -442,46 +443,59 @@ export class PersonService {
     return true;
   }
 
-  async mergePerson(auth: AuthDto, id: string, dto: MergePersonDto): Promise<BulkIdResponseDto[]> {
+  async mergePerson(auth: AuthDto, id: string, dto: MergePersonDto): Promise<MergePersonResponseDto> {
     const mergeIds = dto.ids;
+    if (!(mergeIds instanceof Array)) {
+      throw new BadRequestException('ids is not an array');
+    }
     await this.access.requirePermission(auth, Permission.PERSON_WRITE, id);
-    const primaryPerson = await this.findOrFail(id);
+    let primaryPerson = await this.findOrFail(id);
     const primaryName = primaryPerson.name || primaryPerson.id;
 
     const results: BulkIdResponseDto[] = [];
 
     const allowedIds = await this.access.checkAccess(auth, Permission.PERSON_MERGE, mergeIds);
 
-    for (const mergeId of mergeIds) {
-      const hasAccess = allowedIds.has(mergeId);
+    for (let i = 0; i < mergeIds.length; i++) {
```

Comment:
Why did you change this from being a for each? I don't see any use of an index

Reply:
You want this smart merge to apply only for the first person and if you only merge one person

Reply:
Oh I see, I've overlooked 472. My bad!

Reply:
I think you could use `for (let i in mergeIds` though

Reply:
How to get the index then ? It feels easier to use `for (let i = 0; i < mergeIds.length; i++)` than `for (let i in mergeIds)` when you need the index

Reply:
`for (let i in mergeIds)` != `for (const id of mergeIds)`. `in` gives you the index (i) whereas `of` is a for each

Reply:
What's the benefit over `for (let i = 0; i < mergeIds.length; i++) ` ? It's faster and you can easily have the object with the current index

Reply:
I only use `for (const id of mergeIds)` when I don't need the index because it's easier to maintain even if it's slower

Reply:
Yeah you're right I wrote bullshit. Sorry :/

Reply:
ahaha don't worry, I'm always curious if there's a new way to do things better :smile: